src/oracles/CoinbaseOracle.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.17;
+
+import {IAggregatorV3Source} from "../interfaces/chainlink/IAggregatorV3Source.sol";
+
+/**
+ * @title CoinbaseOracle
+ * @notice A smart contract that serves as an oracle for price data reported by a designated reporter.
+ */
+contract CoinbaseOracle is IAggregatorV3Source {
+    address immutable reporter;
+
+    uint8 public immutable decimals;
+
+    string public symbol;
+
+    uint80 public lastRoundId;
+
+    mapping(uint80 => int256) public roundAnswers;
+    mapping(uint80 => uint256) public roundTimestamps;
+
+    /**
+     * @notice Emitted when a new price is pushed.
+     * @param roundId The round ID of the new price.
+     * @param price The price that was pushed.
+     * @param timestamp The timestamp at which the price was pushed.
+     */
+    event PricePushed(uint80 indexed roundId, int256 price, uint256 timestamp);
```

Comment:
This is non-standard event commenting. I dont think we do it like this elsewhere.

Reply:
+1